kotlinx-coroutines-core/jvm/src/internal/MainDispatchers.kt

```diff
@@ -53,13 +58,29 @@ public fun MainDispatcherFactory.tryCreateDispatcher(factories: List<MainDispatc
     try {
         createDispatcher(factories)
     } catch (cause: Throwable) {
-        MissingMainCoroutineDispatcher(cause, hintOnError())
+        createMissingDispatcher(cause, hintOnError())
     }
 
 /** @suppress */
 @InternalCoroutinesApi
 public fun MainCoroutineDispatcher.isMissing(): Boolean = this is MissingMainCoroutineDispatcher
 
+// R8 optimization hook, not const on purpose
+@Suppress("MayBeConstant")
+private val SUPPORT_MISSING = true
```

Comment:
It's better to report this case to R8 instead of workarounding it here, especially with new SL optimization

Reply:
Maybe @wojtek-kalicinski can verify that missing dispatcher is correctly removed when R8 optimizes SL?

Reply:
It is not removed now, but that is totally due to a different bug in R8. This constant cannot be made `const` anyway, because if it is `const` then it leaves no trace in the class files, making it impossible to configure via R8.

Reply:
I'm leaving this `SUPPORT_MISSING`. The logic around it seems to bee too complex for R8 to analyze, but with `-assumenosideeffects class kotlinx.coroutines.internal.MainDispatchersKt { boolean SUPPORT_MISSING return false; }` I can verify that it gets removed.